pkg/components/prometheus-operator/component.go

```diff
@@ -57,11 +57,11 @@ type Grafana struct {
 // Prometheus object collects sub component Prometheus related information.
 type Prometheus struct {
 	MetricsRetention            string            `hcl:"metrics_retention,optional"`
-	ExternalURL                 string            `hcl:"external_url,optional"`
 	NodeSelector                map[string]string `hcl:"node_selector,optional"`
 	StorageSize                 string            `hcl:"storage_size,optional"`
 	WatchLabeledServiceMonitors bool              `hcl:"watch_labeled_service_monitors,optional"`
 	WatchLabeledPrometheusRules bool              `hcl:"watch_labeled_prometheus_rules,optional"`
+	Ingress                     *types.Ingress    `hcl:"ingress,block"`
```

Comment:
I'm not a fan of adding an option, which is insecure and there is no way to secure it.

Reply:
I am not sure how we can enable federation without it. Specially when the clusters are in different region.

Reply:
We could at least enable basic HTTP auth to protect it. Not ideal, but something.

Reply:
But how does federation work in that case? Prometheus does not support any authentication when scraping.

Reply:
Can't you use the `basic_auth` section? See https://prometheus.io/docs/prometheus/latest/configuration/configuration/

Reply:
So if we were to use basic authentication then it has to be at the ingress layer because Prometheus does not support it:

> Prometheus does not directly support basic authentication (aka "basic auth") for connections to the Prometheus expression browser and HTTP API.

src: https://prometheus.io/docs/guides/basic-auth/

Now to enable an authentication at the ingress level Contour only supports **TLS Client authentication** and that too with `HTTPProxy` object and not `Ingress` object. So if we were to switch from `HTTPProxy` to `Ingress` then we need to make sure that the cert-manager works with that object the way it does with inbuilt `Ingress`.

contour client cert authn: https://projectcontour.io/client-cert-auth-ingress-improvements/
prometheus client cert authn: https://prometheus.io/docs/prometheus/latest/configuration/configuration/#tls_config